### ISSUE TITLE
Dark Souls III performance regression with Proton 3.16 (374320)

### ISSUE THREAD

**Issue by Bloodaxe95** [ValveSoftware/Proton#478]
Game runs perfectly on my end, I notice no difference between Linux and Windows. Both standard mouse-keyboard and the Steam Controller work flawlessly!

- OS: Xubuntu 18.04
- GPU/driver-version: GTX 970 - nvidia-driver-396
- Kernel: 4.15.0-32-generic
- Proton: 3.7

**Comment by Bednar87** [comment]
crashes for me when I click new game.

`Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:         Intel(R) Core(TM) i5-3570K CPU @ 3.40GHz
    CPU Family:  0x6
    CPU Model:  0x3a
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3800 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.18.4-arch1-1-ARCH
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12001000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  <Runtime disabled>

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 970/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 396.54
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x10de
    DeviceID:  0x13c2
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1200
    Desktop Resolution: 1920 x 1200
    Primary Display Size: 20.39" x 12.76" (24.02" diag)
                                            51.8cm x 32.4cm (61.0cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 4096 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: VIA VT2020

Memory:
    RAM:  7938 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_GB.utf8
    Total Hard Disk Space Available:  104481 Mb
    Largest Free Hard Disk Block:  26759 Mb
    VR Headset: None detected

Recent Failure Reports:
`

gist: https://gist.github.com/Bednar87/e71fb272481c8f64efc7ee1b614218e3

**Comment by ghost** [comment]
@Bloodaxe95 please do further testing with a family shared dummy steam account. Playing Dark Souls 3 using DXVK might trigger the anti-cheat system and cause a softban.

**Comment by photex** [comment]
Gets to a point where I get a mouse cursor then exits without an error

**Comment by meowmeowfuzzyface** [comment]
This game was working flawlessly for me on Friday.  Today, it is not launching either with 3.73 or 3.74 beta, with or without "PROTON_USE_WINED3D11=1 %command%".

I'm on Ubuntu 18.04 with the 18.3.0-devel - padoka PPA driver for Fury X, same as Friday.

Possibly related, Dark Souls 2 was also working on Friday, but today it crashes unless launched with "PROTON_USE_WINED3D11=1 %command%".

**Comment by Emanem** [comment]
Main issues for me:

* In order to adjust the screen to 1440p in fullscreen, one has to go to window mode, then change to 1440p, then back to fullscreen - this setting doesn't get saved apparently 
* All maxed out with 1080 GTX (396.54) and latest Proton (3.7-4 beta) the FPS is _only_ 37.5 in 3440x1440 full screen; if I run in 1080p or 1440p windowed it's 60 FPS
* Sometimes the game doesn't exit properly and needs to be killed (when changing resolution form 1080p to 1440p)
* Screensaver/lock gets activated when playing with pad only

Almost _whitelist_... the game is very playable with my PS3 pad and it's just fantastic... (Ubuntu 16.04.4 with HWE).

**Comment by kayosiii** [comment]
The issue I am having is that Steam controller will look but not move or attack, ditto for keyboard and mouse

**Comment by meowmeowfuzzyface** [comment]
Not sure what happened, but the game is now working flawlessly.  I haven't changed anything so I guess there was a Proton update?  I'm using the PROTON_USE_WINED3D11=1 %command% launch option.

Ubuntu 18.04 with the 18.3.0-devel - padoka PPA driver for Fury X.

**Comment by michaeltintiuc** [comment]
I get a crash each time I try to open the door after Iudex Gundyr, I am however running this on 390 nvidia drivers, this _could_ be the issue :man_shrugging:

**Comment by dittr** [comment]
@michaeltintiuc do you have nvidia-driver-396, i had the same issue but updating the nvidia driver fixed it for me. (with nvidia-390 i got the same crash)

**Comment by michaeltintiuc** [comment]
I'm running Solus OS and the devs are kind enough to give some extra testing to the 396 drivers :laughing:  and I'm waiting patiently, but that's awesome to know @Sw3eM , thanks for letting me know!

**Comment by Xeers** [comment]
Reborn OS (Arch Linux)
AMD FX, Nvidia Geforce GTX 950
The game works perfectly but I can not upload the resolution more than 1920x1080 (my monitor is 2560x1080)

The command PROTON_USE_WINED3D11 = 1% command% is not necessary, it breaks the game

**Comment by UndeadKernel** [comment]
@Bednar87, did you figure out a solution to your problem? I have the same specs and problem (game crashes if I choose "new game").
To solve the issue, I started the game in Windows and saved as soon as I had control. Afterwards,I copied the save game from windows to linux. This way, I don't have to choose "new game", just "continue" and the game works rather well.

**Comment by Bednar87** [comment]
@UndeadKernel, nope. I don't have access to a windows machine I can try this on, but thanks for reporting you have found a solution that works for you. 

Going to google to see if save games are portable...

**Comment by meowmeowfuzzyface** [comment]
@Bednar87 Save games are portable.  For both DS2 and 3, the save directory in Windows is under C:\Users\<your user>\AppData\Roaming\.  The folder is called DarkSoulsIII.

You can copy paste that folder into /steamapps/compatdata/374320/pfx/drive_c/users/steamuser/Application Data/.

**Comment by patlefort** [comment]
I'm on Antergos. I have a problem with this game and my dualshock 4 controller: The right trigger isn't working properly. It does normal attacks and if I try to bind something to it, it's not detected. It works in DS1 and the left trigger work. Very strange problem. I tried to rebind it with steam controller settings but it doesn't work.

Next small problem is sometimes the cursor will become visible on the middle of the screen. To hide it, I have to press start, move the mouse, close the menus.

Another big problem: Sound. Sometimes, all the sounds will sound like it's coming from the front-left on my 5.1 setup. Restarting the game fix it until it does it again. The problem also occur in DS1.

**Comment by jonlap** [comment]
Seems to run very well (I only played 10-15 minutes though). Only issues were that problem with 1440p resolution mentioned earlier and Xbox 360 controller analog sticks and dpad being overly sensitive (working normally with every other game) to the point game is practically unplayable with it. Info [here](https://gist.github.com/skry/4e2fd229ffb108d565dce6737952dd08).

**Comment by GloriousEggroll** [comment]
Game runs fine here on DXVK:
Proton: 3.7 Beta 6
OS: Arch
Kernel: 4.19.0-rc2
Mesa: Mesa 18.3.0-devel **(git-14fe9fa11b)**
LLVM 7.0
GPU: AMD RX Vega 64
CPU: AMD Threadripper 1950x

**Comment by tamato** [comment]
@GloriousEggroll What do you mean "on DXVK"?

What would I need to do to get it running on DXVK?

**Comment by UndeadKernel** [comment]
@Bednar87, I believe I've figured out why the game crashes when selecting "New Game".
The backtrace of the crash shows that a problem lies in the Bink video player:
     `0 0x00000000002ded63 in binkw64 (+0xed63) (0x00007fb815988da0)`

To fix the "New Game" crashing issue, in the folder `Steam/steamapps/common/DARK SOULS III/Game/movie` rename the file `fdp_opening.bik` to `fdp_opening.bik.bkp`. This way, the bink libraries won't find the video and will just skip it, allowing you to play the game.

Of course this means that you won't see the introduction video. If you want to see it, just open the renamed video file with VLC. To listen to the narrator, change the audio channel.

Anybody knows what is needed to adequately run bink videos in wine? It seems that the DLL file `binkw64.dll` is causing problems.

**Comment by darkpumpkins** [comment]
It does crash for me to at startup. 
I was previously able to play it outside steam linux on wine + dxvk 0.63.

Proton: 3.7
OS: Ubuntu 18.04.1
Kernel: 4.15.0-34-generic
Mesa: Nvidia 396.54
GPU: Nvidia MSI GTX 970 Gaming 4G
CPU: Intel core i5-2500k

[steam-524220.log](https://github.com/ValveSoftware/Proton/files/2380955/steam-524220.log)

**Comment by kisak-valve** [comment]
Hello @darkpumpkins, your log doesn't match this game. Was this issue report where you intended to post?

**Comment by darkpumpkins** [comment]
Crash on startup ( this time with the correct log :P ), well the game does not start at all.

Proton: 3.7-6
OS: Ubuntu 18.04.1
Kernel: 4.15.0-34-generic
Mesa: Nvidia 396.54
GPU: GeForce GTX 970/PCIe/SSE2
CPU: Intel® Core™ i5-2500K CPU @ 3.30GHz × 4 

[steam-374320.log](https://github.com/ValveSoftware/Proton/files/2380995/steam-374320.log)

**Comment by kisak-valve** [comment]
`err:winediag:FILE_CreateFile Too many open files, ulimit -n probably needs to be increased` is the line of interest in the log.

**Comment by darkpumpkins** [comment]
great !
Dark souls III run !
Thanks for the tutorial, ulimit also was a pain for my samba ... -_- 
where should i put my old windows saves into steam proton to restore my old saved games ?

**Comment by krompus** [comment]
Game is running stable here, but performance takes a hit compared to Windows; my FPS is halved (30ish on Linux, smooth 60 on Windows):

- Arch Linux
- i5 4670K @3.8GHz
- GTX 770 Nvidia 396.54-4
- Proton 3.7-6

Hopefully future updates will improve performance. I'm about due for a new rig, though. :)

**Comment by michaeltintiuc** [comment]
It seems to have gotten worse, it no longer works, unless I rename the video files and the framerate took a hit.

**Comment by darkpumpkins** [comment]
if someone else does encounter problems with ulimit here is a solution : https://phpsolved.com/ubuntu-16-increase-maximum-file-open-limit-ulimit-n/

**Comment by krompus** [comment]
@darkpumpkins ah, thanks; I didn't know this game was affected by that issue. I discovered this when testing Downwell a few weeks ago: crashed after about 60 seconds before, stable as a rock after raising the limit.

**Comment by darkpumpkins** [comment]
@krompus  kisak-valve mention me a point that i did not notice about steam play into the requirements => https://github.com/ValveSoftware/Proton/blob/proton_3.7/PREREQS.md#fd-limit-requirements
I had the ulimit error into Nioh and Dark souls III but for Nioh seems wine related problem ^^.

Anyone noticed better performances with lastest nvidia beta driver ( 396.54.05 ) ? ( here is the repository : https://launchpad.net/~graphics-drivers/+archive/ubuntu/dev )

**Comment by patlefort** [comment]
I fixed my controller problem by applying some community controller profile. I don't know why this worked.

**Comment by meowmeowfuzzyface** [comment]
@patlefort If you have Steam configuration for your controller on by default, try turning it off and letting the game recognize the controller directly.  You can do this per-game in Big Picture mode by going to the game in your library and selecting Manage Game. Set the controller configuration to Forced Off.

Steam controller configuration doesn't work with any of my Proton games, so this is how I've been working around it.

**Comment by jomarocas** [comment]
i have a gtx 970 and latest drivers and run in certain places in 30 fps and others in 50, this is stable in windows with 60 fps

**Comment by Emanem** [comment]
> i have a gtx 970 and latest drivers and run in certain places in 30 fps and others in 50, this is stable in windows with 60 fps

Are you playing fullscreen or windowed? Because fullscreen incurs in a Nvidia drivers problem where it's basically forcing V-sync and then it goes back to 30 fps. For example, I have a 1080 GTX and I play 21:9@1440p and windowed it goes around 60 FPS, but fullscreen it goes to 37.5 (my screen is 75 Hz) because it can't precisely hit the 60 FPS with V-sync like enabled (i.e. even 59 FPS is not good enough for V-sync).

**Comment by Niroc** [comment]
Whilst in Fullscreen mode, I'm unable to change the resolution from 1080p down to my normal 900p res like in Windows. Swapping to Windowed mode, changing the resolution and swapping back to Fullscreen mode doesn't not work either which people have been doing to go to 1440p. For me it just defaults back to 1080p

On another note, whilst Fullscreen my fps is NOT locked to 30 as mentioned above by @Emanem. I get not so smooth 38-47 fps but, to maximize performance and resource I use a dedicated X session only running the openbox window manager & Steam on my NVIDIA card and V-sync is not enabled on this X session. My normal desktop runs on my Intel GPU on another TTY.

Also, I've been playing for 10 hours with no ban yet from the anti-cheat system.

- GPU: GTX 870m 6GB 
- CPU: I7-4710MQ @ 3.5 GHz 
- OS: Manjaro
- Driver/LLVM version: Nvidia 396.54 
- Kernel version: 4.14.71-1
- Proton version: 3.7-7 Beta

**Comment by Niroc** [comment]
The resolution issue should be fixed in the following commit: https://github.com/doitsujin/dxvk/commit/87f1cd238516bc76d9525a1f75314a27d6e53e68

We'll need to wait for DXVK v0.82 to see it in proton most likely, unless Valve cherry pick the commit when updating the v0.81

**Comment by UndeadKernel** [comment]
With the latest Proton Beta (3.7-7), I'm observing drastic FPS drops in many areas where I would not previously have FPS problems. From a stable 45-50 FPS, the FPS drops to 20 and below, making the game unplayable.

Reverting back to Proton 3.7-6 is the only way to keep a stable FPS. 

My system information can be found [here](https://pastebin.com/raw/qax1xrEh).

**Comment by kisak-valve** [comment]
Hello @UndeadKernel, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Niroc** [comment]
@kisak-valve I think the regression is down the temporary high CPU load from the DXVK State Cache. @UndeadKernel could try starting the game with following launch option "DXVK_STATE_CACHE=0" to check. 
Proton 3.7-7 works fine for me but, I've been pretty much everywhere in-game.

**Comment by UndeadKernel** [comment]
@Niroc, I tried playing with the launch option "DXVK_STATE_CACHE=0" and still saw FPS drops. It seems to be related to the multiplayer's mode though: I notice considerable FPS drops only after I traveled to another player's "map". The map, though, was the same as mine. For the first 30 minutes of play (without multiplayer), the game was performing as expected. Therefore I do not believe it's related to the cache.

**Comment by Niroc** [comment]
@UndeadKernel peculiar... I'll do some invading too see if my laptop does the same. Check out nvidia-xrun & Feral game mode (for Feral game mode set you PAM file "/etc/security/limits.conf" too "*\*insertUserNameHere\**            -       nice            -5 " then restart, The nice value does seem to help with the responsiveness of my gamepad. ) I'll let you know what I find.

**Comment by UndeadKernel** [comment]
@Niroc, I was already using Feral's Game Mode and my system has been configured with the necessary changes (i.e., ulimit and nice limits) to run Proton.
With Proton 3.7-7, the game's window is managed differently: I can use ALT+F4 to kill the game. With 3.7-6 ALT+F4 does nothing. I don't know if this is making KDE (my window manager) change its compositor settings for the game.

**Comment by krompus** [comment]
I'm getting massive FPS drops when I warp at bonfires on 3.7-7 stable and 3.7-8 beta; 3.7-6 stable didn't have this issue. I was using it just last night, and the only software that has changed since has been 3.7-7 replacing 3.7-6.

- i5 4670K @4.0GHz
- GTX 770 2GB with nvidia 410.57-4
- Arch Linux
- Linux kernel 4.18.12.arch1-1 

Edit: installed nvidia-vulkan-dkms 396.54.09, and the issue persists; not a driver issue. 720p works pretty well despite looking like poo, so that's what I'm using for now.

**Comment by Niroc** [comment]
@Undead Cool, I did some invading btw and I haven't noticed any frame drops on the FPS counter when in other other peoples worlds. Personally, I use the lightweight Openbox window manager that literally just runs steam & my games on it's own X session on my Nvidia card. I haven't touched KDE this decade so, I couldn't say about that much about that I'm afraid...

I haven't noticed the bonfire warping frame drop outlined by @krompus, I run the game at 900p on my GTX 870m with most setting set to med/low to ensure a consistent 60 FPS (Same as Windows as, the second DLC is rather GPU taxing in certain areas). More details on my system are 9 posts up though, I'm now running Proton 3.7-8 beta.

I have encountered a new issue were the audio begins to crackle after running Dark Souls 3 for a prolonged period. (It took 5+ hours to happen, it's happened more than 3 times)

**Comment by UndeadKernel** [comment]
Reporting on the FPS drop issue after testing with Proton beta (3.16-1). The in-game performance is generally much much better. I've seen an FPS increase of more than 20%.
However, after traveling to another world (either through a bonfire or through an invasion), the FPS drops to less than 15 FPS and the game becomes unplayable.

Any idea how I could debug this issue?

**Comment by meowmeowfuzzyface** [comment]
Dark Souls 3 (and 1 and 2) are crashing on launch for me since the Proton 3.16 update.

-Name of the game with compatibility issues: Dark Souls PTDE, Dark Souls II Scholar of the First Sin, and Dark Souls III

-Steam AppID of the game: 211420, 335300, and 374320

-GPU: Radeon Fury X

-Driver/LLVM version: mesa 18.3.0-devel from Padoka PPA, LLVM 8.0.0

-Kernel version: 4.15.0-36-generic

-Link to full system information report: Gist

-Proton version: 3.16 and 3.7-8

[steam-211420.log](https://github.com/ValveSoftware/Proton/files/2480721/steam-211420.log)
[steam-335300.log](https://github.com/ValveSoftware/Proton/files/2480722/steam-335300.log)
[steam-374320.log](https://github.com/ValveSoftware/Proton/files/2480723/steam-374320.log)

Description:
All 3 games were working perfectly with 3.7-8 until the 3.16 update. Switching back to 3.7-8 does not solve this crash. The behavior is the same for both versions.

Since the update, Dark Souls PTDE, Dark Souls II Scholar of the First Sin, and Dark Souls III all crash on launch. Other DX11 games are working fine with DXVK (Witcher 3, Skyrim SE, Killing Floor 2).

Oddly, Dark Souls II and III do not crash if they are minimized (using alt + pgdn). Upon maximizing, they are stable if there is no input. If I press any button, the game crashes after one or two seconds. Dark Souls PTDE crashes even when minimized.

Things I have tried:
Using the launch option for WINED3D11 or NO_D3D11.
Using the -fullscreen launch option.
Deleting the WINE prefix.
Verifying integrity of the game files.
Disabling the compositor.

Reproduction:
For all 3 games: Launch the game normally. A window will pop up and crash without displaying anything.

For Dark Souls 2 and 3: Launch the game and immediately minimize the window. The game will launch normally in the background. Upon maximizing the window, press any button. The game will crash after 1-2 seconds.

**Comment by michaeltintiuc** [comment]
@meowmeowfuzzyface I had previously had similar issues with other games, reinstalling them seemed to help. I have just tried swtiching to 3.16 and if anything the performance had improved.
Running on Solus OS (kernel 4.9.113), GTX 1050Ti (drivers 410.57)

**Comment by michaeltintiuc** [comment]
@meowmeowfuzzyface also check the [changelog](https://github.com/ValveSoftware/Proton/wiki/Changelog) for the changed launch options

**Comment by meowmeowfuzzyface** [comment]
@michaeltintiuc Thanks for the heads up.  Unfortunately, reinstalling did not change the behavior.  It is still crashing to desktop briefly after launching.  Using the new PROTON_USE_WINED3D command causes an instant crash without the Dark Souls III window even appearing at all.

**Comment by meowmeowfuzzyface** [comment]
I ppa-purged the Padoka unstable branch and upgraded to Ubuntu 18.10 which uses mesa 18.2.2.  The crashing has gone away with 18.2.2.  So this is an issue being caused by the recommended driver version (18.3.0-devel).

**Comment by mlassnig** [comment]
Not crashing, but still experiencing the FPS drops. What is weird is that I get stable 60fps, after first warp/YD I get "stable" 30fps, after the second  I get stable 15 fps, third warp 8fps, then I get disconnected. Looks like a bug somewhere in the way vsync is applied, force-halfing the FPS in some weird way?

--
Kernel 4.18.6, GTX970, NVidia 396.54.09, Vulkan 1.1.70, Proton 3.16-3

**Comment by UndeadKernel** [comment]
@mlassnig, some time ago @krompus discovered that you can fix the FPS drops by going into windowed mode (ALT+RET) and switching back to full screen mode (ALT+RET). The only downside is that if you switch modes too fast, you crash the game :D

**Comment by Brisse89** [comment]
Put in more than 70 hours of DSIII using Proton now. For the most part it works well with a solid 60fps, but occasionally I get the frame-rate dips (they are quite severe and game becomes unplayable) that others are reporting. They seem to happen randomly after loading screens such as when warping between bonfires. Sometimes I can play hours before it happens, sometimes just minutes. My solution has been to restart the game but I'm going to try ALT+RET as suggested above next time I encounter the issue.

https://gist.github.com/Brisse89/a34a926d26c53bc7c4d279d1911b8fa5

**Comment by mlassnig** [comment]
@UndeadKernel @Brisse89 ALT+RET worked for me.

**Comment by meowmeowfuzzyface** [comment]
Do you guys get this occasional frame drop issue if you play in windowed mode? Or is it only when in fullscreen? I've been playing in windowed with no problems after downgrading to mesa 18.2.2 to solve the crash I was experiencing earlier. I will test in fullscreen to see if I get it too.

**Comment by Brisse89** [comment]
I'm getting the frame-drops right now as we speak. ALT+RET does not help for me. Frame-rate in windowed mode is even worse.

top barely shows any cpu usage but radeontop shows 'Graphics pipe 100,00%' constantly despite the low frame-rate.

**Comment by Brisse89** [comment]
```
              radeontop unknown, running on FIJI, 120 samples/sec               
                                         │
                   Graphics pipe 100,00% │                                     
─────────────────────────────────────────┼──────────────────────────────────────
                    Event Engine   0,00% │
                                         │
     Vertex Grouper + Tesselator  14,17% │     
                                         │
               Texture Addresser  44,17% │                
                                         │
                   Shader Export  44,17% │                
     Sequencer Instruction Cache  11,67% │    
             Shader Interpolator  97,50% │                                    
                                         │
                  Scan Converter  98,33% │                                    
              Primitive Assembly  14,17% │     
                                         │
                     Depth Block  95,83% │                                   
                     Color Block  40,00% │              
                                         │
                                         │
                                         │
                                         │
                                         │

```

**Comment by UndeadKernel** [comment]
@Brisse89, for it to work, I needed to go back to full screen after switching to windowed mode. So, just issue another "ALT+RET" to go to full screen and let us know if that fixed it for you.

**Comment by Brisse89** [comment]
@UndeadKernel Tried several times but it was not helping.

**Comment by Brisse89** [comment]
Restarted the game, got the frame-rate dip again, but this time it did help to switch back and forth. Wierd!

Edit: Seems a bit random. Sometimes it works, other times I still have to quit and restart.

**Comment by Brisse89** [comment]
Is it just me or does the frame-rate dip happen a lot more often on 3.16-3 beta than 3.7-8. I just switched back to 3.7-8 and I could play for hours without much issue. I feel like with 3.16-3 beta it happens every second or third loading screen which is quite frustrating.

**Comment by Fubs** [comment]
I'm also getting the same frame rate dips after dying/warping that @Brisse89 described. For me though, it happens EVERY TIME I die or warp. I suspect it has something to do with vsync as @mlassnig explained, since I also get the halving effect after each death/warp. Anyone got a proper fix for this yet?

**Comment by TheSHEEEP** [comment]
Same here, I had to switch back. It definitely happened with one of the latest 3.16-X versions. I went back to 3.7-8 and it works fine now.

**Comment by HonkingGoose** [comment]
### Description of issue:

Using Steam from within the terminal registers a buffer overflow for Dark Souls 3.

### System Information
- OS: Ubuntu 18.04.1 LTS
- GPU: Nvidia GeForce GTX 780 Ti
- Driver/LLVM version: Nvidia 410.73
- Kernel version: 4.15.0-38-generic
- Link to full system information report as [Gist](https://gist.github.com/HonkingGoose/593bd21e8affb816fe86ad9df06b35d5).
- Proton version: 3.16-4 Beta
- Proton log following steps to reproduce: [steam-374320.log](https://github.com/ValveSoftware/Proton/files/2545170/steam-374320.log)
- Controller used: Wireless controller for XBOX One, model 1708, used wired via USB cable.

### Steps to reproduce Proton log:
1. Start Steam client from within terminal.
1. Start Dark Souls III by clicking on run.
1. View the intro splash screens.
1. Press A on "Press any button" prompt.
1. Press A on "Patch notes information" prompt.
1. Press D-pad up to select Quit.
1. Press A to exit game. (Quit)
1. Exit Steam.

### Output of the program, when run from the terminal:
```
Running Steam on ubuntu 18.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
[2018-11-03 18:08:41] Startup - updater built Nov  2 2018 23:16:17
[2018-11-03 18:08:41] Opted in to client beta 'publicbeta' via beta file
You are in the 'publicbeta' client beta.
[2018-11-03 18:08:41] Verifying installation...
[2018-11-03 18:08:41] Verification complete
*** buffer overflow detected ***: Z:\home\giraffe\.steam\steam\steamapps\common\DARK SOULS III\Game\DarkSoulsIII.exe terminated
[2018-11-03 18:09:50] Shutdown
```

### Discussion:
So it says: **buffer overflow detected**, but following the steps to reproduce just enables me to close the game as normal. The game itself doesn't crash on me.

**Comment by ArsenArsen** [comment]
## System Information
- GPU: GTX 1050Ti
- Driver/LLVM version: nvidia 396.54
- Kernel version: 4.17.19
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/ArsenArsen/ce236586d14ed31796d02bd1a37b488e
- Proton version:

## Symptoms
The main menu and the hud are in the top left quadrant of the screen (image attached)

## Reproduction
Start the game

![screenshot 2018-11-06 07 11 53](https://user-images.githubusercontent.com/7805050/48046246-fca64f80-e193-11e8-9fb2-30af425307a4.png)

**Comment by leifmetcalf** [comment]
After a while Ubuntu detects that I haven't jiggled my mouse in a while and the screen fades black and locks. Is there an option to stop turning the screen off when a fullscreen game is open?

**Comment by darkpumpkins** [comment]
@leifmetcalf  on gnome 3 there is the caffeine extension to prevent such a behaviour => https://extensions.gnome.org/extension/517/caffeine/

**Comment by krompus** [comment]
@leifmetcalf you can also try "joystickwake" to delay screensaver activation with joypad activity: https://launchpad.net/~foresto/+archive/ubuntu/toys

**Comment by kisak-valve** [comment]
Dark Souls III low fps drops since 3.16 

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1942.
@icemanx7 posted on 2018-11-10T06:55:51:

Using proton 3.7.8 I get full 60fps on Dark Souls 3 but using the latest 3.16.4 beta I get drops to about 30fps on with both settings on max.

GPU: GTX 1060
CPU: 2400g
RAM 16G
OS: Arch Linux
Nvidia Drivers: 410.66

**Comment by doitsujin** [comment]
Is there any reliable way to reproduce the FPS drops? I did try dying a couple of times and teleporting between bonfires, but frame rates remained stable on my RX 480.

**Comment by icemanx7** [comment]
@doitsujin  Well basically just start a new character and try to get a view of the fountain in the starting area that tanks my fps on the new proton version. Also when fighting the first boss just turning your camera to face that door that leads to firelink shrine. Inside firelink stand in the middle and look towards Andre that also tanks my fps to 30 directly all 3 times. But on 3.7 it's not there. Very stable.

I also installed the Nvidia beta drivers with transform feedback and still the same thing happens.

**Comment by doitsujin** [comment]
Hm, this is weird. I just tested all three locations and FPS remain at 60 at all times for me.

**Comment by icemanx7** [comment]
@doitsujin what cput do you Have?  I'll try a different OS like Ubuntu maybe it's something to do with Arch.

**Comment by krompus** [comment]
@icemanx7 It would surprise me if it's an issue with Arch itself, but I'll set up a buntu install as well, for testing.

**Comment by leifmetcalf** [comment]
I used to have these fps drops when changing locations about a week ago, but in the last few days (≈9 hours of play) I haven't had them at all. Ubuntu 18.10, Proton 3.16-4, & Nvidia 410.73.

**Comment by doitsujin** [comment]
@icemanx7 I'm on Arch as well, so that shouldn't be an issue per se. Just make sure that you have the [FD limit](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements) set up correctly so that esync works properly.

My CPU is a Ryzen 2700X, but I did re-test it in powersave mode (2.2 GHz) and found no noticable slowdowns either. There also should't really be any VRAM related issues even on lower-end cards.

**Comment by icemanx7** [comment]
I upgraded to Arch testing repos to get the new Nvidia 410.73 drivers to see if it would make a difference and it doesn't. So I doubt switching to Ubuntu would yield different results. Oh I am using plasma as my DE if that info would help.

**Comment by kisak-valve** [comment]
Hypothetically, what if the AMD apu was being used with vulkan instead of the nvidia card, @icemanx7, can you look in ~`/etc/share/vulkan/icd.d`~ `/usr/share/vulkan/icd.d` for `radeon*.json` files and temporarily move them somewhere safe.

**Comment by icemanx7** [comment]
> Hypothetically, what if the AMD apu was being used with vulkan instead of the nvidia card, @icemanx7, can you look in `/etc/share/vulkan/icd.d` for `radeon*.json` files and temporarily move them somewhere safe.

I don't have a /etc/share folder

**Comment by icemanx7** [comment]
> Hypothetically, what if the AMD apu was being used with vulkan instead of the nvidia card, @icemanx7, can you look in `/etc/share/vulkan/icd.d` `/usr/share/vulkan/icd.d` for `radeon*.json` files and temporarily move them somewhere safe.

There isn't a radeon*.json file in that folder only nvidia_icd.json.

**Comment by Brisse89** [comment]
> There isn't a radeon*.json file in that folder only nvidia_icd.json.

You probably don't have the [radeon vulkan driver](https://www.archlinux.org/packages/extra/x86_64/vulkan-radeon/) installed which means DXVK can only run on your Nvidia GPU.

**Comment by Brisse89** [comment]
> @icemanx7 I'm on Arch as well, so that shouldn't be an issue per se. Just make sure that you have the [FD limit](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements) set up correctly so that esync works properly.

This seems to have fixed the framerate drops after warping between bonfires/worlds for me.

Edit: No, I was wrong. Issue came back after playing for about 2 hours.

**Comment by icemanx7** [comment]
I don't think that's the same problem I am having. When I warp my performance stays okay but when I view certain areas my fps just tanks to 30fps immediately. Like when I am in firelink shrine looking at andre it tanks but when I turn around it jumps back to 60fps. It is quite annoying because it sometimes happens when I just run around. My fps will 60fps then I will get these 30fps dips and slowdowns.

**Comment by Brisse89** [comment]
@icemanx7 Yes I agree it sounds like two separate issues. I'm not seeing any performance drops like those you describe, nor did I see them before fixing the FD limit.

**Comment by icemanx7** [comment]
What is weird is on 3.7.8 I get perfect performance. But thanks anyway for the help guys. If anyone has any other suggestions let me know.

**Comment by krompus** [comment]
Alright, just set up a fresh Ubuntu 18.10 install, with Nvidia Vulkan Beta drivers 396.54.09 for my GTX 770, and I'm getting the same FPS drops upon bonfire warp/respawn, with all Proton versions (3.7-8, 3.7-8 beta, 3.16-4 beta); it's not an issue with Arch. If anything, I'm getting generally higher and more consistent FPS on my Arch install.

**Comment by icemanx7** [comment]
@krompus thanks for that testing same situation with me. I also still get drops on Ubuntu and on 3.7.8 but on 3.7 the drops isn't so bad compared to 3.16.4. I am trying to test DS3 with lutris to see if is a DXVK problem but I can't get the game to run.

**Comment by v0idwalker** [comment]
> Alright, just set up a fresh Ubuntu 18.10 install, with Nvidia Vulkan Beta drivers 396.54.09 for my GTX 770, and I'm getting the same FPS drops upon bonfire warp/respawn, with all Proton versions (3.7-8, 3.7-8 beta, 3.16-4 beta); it's not an issue with Arch. If anything, I'm getting generally higher and more consistent FPS on my Arch install.

I am getting the same behavior as of now. (there was a bigger update here) The behavior is generally weird, as I do not see a big use of my GPU. (according to the power draw leds)
Yesterday, I was on 60fps stable. 
I am also on arch on the most current kernel(4.19) 
ryzen 2600, 32GB of ram, vega56 with AMDGPU. 
I would guess that this is either something less deterministic, or some distro related problem.  (is mostly just a feeling...)

**Comment by kisak-valve** [comment]
Found in the 2018-11-13 white list update.

Any further issues with this game title should be reported in separate issue reports. This issue report is being held open by the performance regression currently being discussed.

**Comment by leifmetcalf** [comment]
> Alright, just set up a fresh Ubuntu 18.10 install, with Nvidia Vulkan Beta drivers 396.54.09 for my GTX 770, and I'm getting the same FPS drops upon bonfire warp/respawn, with all Proton versions (3.7-8, 3.7-8 beta, 3.16-4 beta); it's not an issue with Arch. If anything, I'm getting generally higher and more consistent FPS on my Arch install.

I just did the same thing, except with version 410.73 nvidia drivers and a GTX 970, and get similar FPS drops following bonfire warps.

**Comment by dragodin** [comment]
### **System Information**
    CPU: AMD Ryzen 5 2600
    GPU: GTX 1060 3GB
    Driver/LLVM version: nvidia 410.73
    Kernel version: 4.18.17
    OS: Manjaro
    Proton version: I tried both 4.7-8 and 3.16-4
    Ulimit Info:
`[xxxxx@xxxxx-pc ~]$ ulimit -Hn`
`1048576`


### **Symptoms**
The game has low fps in general and A LOT of stuttering and dps dips. I play on Low settings, 1080p. I notice that after I teleport a few times with bonfires the performances tank hard, we are talking about 1fps. This is a problem I also had with the native version of Shadow of Mordor where after a few teleports of after I started a mission performances would drop. Some people solved this issue (in shadow of mordor) by adding to the command options "taskset -c 0,2,4 %command%" because, I quote:

> for a AMD CPU command taskset -c 0,2,4 helps a lot
just put it into game launch options in steam like "taskset -c 0,2,4 %command%"
somehow when its not used it switches rapidly between cores and fps drops a lot, with 3 main cores activated only it a lot better, nearly 2x better for me. game seems to only utilize 2 cores anyway.
as for 970, if you put all ultra it runs out of video memory fast and hits the famous 3.5GB bug, everything on high should be enough though and it wont hit the 3.5GB mark. 

Source: https://steamcommunity.com/app/241930/discussions/0/535152276591933829/

I tried with DS3 it but didn't work, the problem persisted. I also tried to disable SMT in the bios but with no results.

EDIT: Now that I think about it I think this is a "Linux" problem, not a "Proton" problem. Because I have the exact same behavior on a native Linux game... I just would like to get to the bottom of this.

**Comment by icemanx7** [comment]
Me too, I actually couldn't believe they whitelisted when I checked this morning. I hope that keeping this issue alive will maybe let it get fixed in the next proton release.

**Comment by dragodin** [comment]
I tried to install Ubuntu 18.4 (from Manjaro), too, with the drivers the latest 415 drivers, and the exact same problems presents itself. Anyone else found a workaround or fix?

**Comment by Jigoku** [comment]
Game runs smooth on max settings... however, everytime i die, as soon as i respawn the game gets stuck in <5fps. After a moment, i get a notification about "unsuitable fps for online mode", and it kicks me to the title screen. 

This happens every time i die.

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD FX(tm)-8320 Eight-Core Processor           
    CPU Family:  0x15
    CPU Model:  0x2
    CPU Stepping:  0x0
    CPU Type:  0x0
    Speed:  3500 Mhz
    8 logical processors
    8 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Linux 4.x (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.19.2
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12003000
    X Window Manager:  awesome
    Steam Runtime Version:  steam-runtime-beta-release_2018-10-24

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1060 3GB/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 415.13
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 144 Hz
    VendorID:  0x10de
    DeviceID:  0x1c02
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 18.90" x 10.63" (21.65" diag)
                                            48.0cm x 27.0cm (55.0cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 3072 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC1150

Memory:
    RAM:  15988 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  468429 Mb
    Largest Free Hard Disk Block:  96865 Mb
    VR Headset: None detected

Recent Failure Reports:


```

**Comment by MatejSpindler** [comment]
The game runs OK with 50 - 60 fps but only for the first time. After game world reloads (if you die and respawn or warp to another location with bonfire) fps drops below 15 fps. The only thing that restores fps is a full restart of the whole game.

System:
```
CPU: AMD Ryzen 7 2700x
GPU: GTX 960 2GB
Driver/LLVM version: Nvidia 410.73
Kernel version: 4.19.2
OS: Arch Linux
```
I tried all currently available proton versions and all have the same problem. Game used to work ok with older version of proton (don't remember exactly but I think it was 4.7-6) which had version of DXVK 0.70.

I tried replacing DXVK in : 
```
"SteamLibrary"/steamapps/common/Proton 3.16/dist/lib64/wine/dxvk
"SteamLibrary"/steamapps/common/Proton 3.16/dist/lib/wine/dxvk
```
Results:
DXVK 0.70 works OK after game reload 
Every version of DXVK after 0.70 up to current 0.92 drops the fps to below 15 fps after a game reload.

This bug is in DXVK and something that changed between 0.70 and 0.71 is causing it.

**Comment by igo95862** [comment]
~~[RX480 card here.](https://gist.github.com/igo95862/4e4c5dace6a494ea57719bff6e7ee34d) No issues so far. From the posts people made I can see correlation between FPS drops and Nvidia GPUs.~~

Seems to be related to low VRAM...

**Comment by doitsujin** [comment]
The problem seems to affect low-VRAM cards and is not trivial to fix. Not sure why 0.70 works fine for some people though.

**Comment by dragodin** [comment]
@doitsujin if it can be of help I tried to run the windows version of Dark Souls III through Lutris wine-staging-3.19, dxvk 0.91 and I had this error, which seems memory related:

`[1110/161635.846:ERROR:angle_platform_impl.cc(54)] populateRenderer11DeviceCaps(906): Error querying driver version from DXGI Adapter.
warn:  D3D11Device::GetDeviceRemovedReason: Stub
System page size: 4096
Could not find page for trampoline in +/- 2GB range of function to hook
Error allocating trampoline memory (no memory within +/-2gb? prior failures?)`

full log:

https://pastebin.com/jMf4URXf

**Comment by Brisse89** [comment]
> [RX480 card here.](https://gist.github.com/igo95862/4e4c5dace6a494ea57719bff6e7ee34d) No issues so far. From the posts people made I can see correlation between FPS drops and Nvidia GPUs.

Negative. I'm having the same issue using R9 Fury.

**Comment by doitsujin** [comment]
@dragodin that message doesn't come from the game, it's from some steam exe.

**Comment by dragodin** [comment]
@doitsujin nevermind then, anyway thanks for your efforts!

**Comment by MatejSpindler** [comment]
@doitsujin I did a git regression test between v0.70 and v0.71 and:
```
f1b075c0f3f01db9f39190a26482e61ff1ce5116 is the first bad commit
commit f1b075c0f3f01db9f39190a26482e61ff1ce5116
Author: Philip Rebohle <philip.rebohle@tu-dortmund.de>
Date:   Mon Aug 27 16:07:38 2018 +0200

    [dxvk] Add CS chunk pool
    
    Reduces the number of dynamic memory allocations for CS chunks by
    recycling them once they are no longer needed. Also fixes a potential
    issue with chunks that are dispatched multiple times.

:040000 040000 0fede36f69aa2a00d697f482cd13d35d830b6c8c e6a27e6afcae76eb225ee4cfe033885153679563 M	src
```

**Comment by dragodin** [comment]
@MatejSpindler is right, overwriting the proton 3.16-4 dxvk folders solves the issue that performance issue after the teleport to a bonfire. I played for an hour and a half with zero issues. I think that the fsp is still a little low though.

Great catch @MatejSpindler !!

**Comment by doitsujin** [comment]
@MatejSpindler hm, interesting. I think I see now why that commit would cause this issue, but unfortunately this means that the *good* performance you were getting before that commit was actually the result of a bug related to deferred contexts. Reverting to the old behaviour is not an option.

**Comment by ArsenArsen** [comment]
Anyone know what can cause the hud to get compressed to top left? Like in: https://github.com/ValveSoftware/Proton/issues/478#issuecomment-436144394

**Comment by drcassar** [comment]
I followed u/lnx-reddit instructions and now I'm able to play DS3 with my 960 with 2GB. I was having the slowdown issue after bonfire teleport.
First I created a file /home/username/dxvk.conf with this line:
dxvk.allowMemoryOvercommit = True
Then added DXVK_CONFIG_FILE=/home/username/dxvk.conf to the launch options
I'm running a fully updated arch and using the "Steam (Runtime)" launcher fo steam.
Link to the u/lnx-reddit post: https://www.reddit.com/r/linux_gaming/comments/9xwxh4/workaround_found_for_dark_souls_iii_performance/e9vyw8v/

**Comment by krompus** [comment]
@drcassar 

> fully updated arch

Which nvidia drivers are you using? I was using nvidia-vulkan-dkms (396.54.09) from the AUR, but it seems to be broken with the latest kernel. I've switched back to main nvidia (currently 410.73), and the workaround you've described hasn't solved the respawn/warp frame drops for me. Using a GTX 770 2GB.

**Comment by igo95862** [comment]
I've tried running game with DXVK_HUD=fps,memory to measure the VRAM usage (assuming it shows the gpu memory) and I've noticed there was almost no difference in memory between low and max textures. (which I would think would have the largest difference in VRAM usage)

**Comment by drcassar** [comment]
> @drcassar
> 
> > fully updated arch
> 
> Which nvidia drivers are you using? I was using nvidia-vulkan-dkms (396.54.09) from the AUR, but it seems to be broken with the latest kernel. I've switched back to main nvidia (currently 410.73), and the workaround you've described hasn't solved the respawn/warp frame drops for me. Using a GTX 770 2GB.

Hey @krompus, I'm using the main nvidia driver. I'm also using no esync, so the launch option is like this: 
PROTON_NO_ESYNC=1 DXVK_CONFIG_FILE=/home/username/dxvk.conf %command%
Are you using the "Steam (Runtime)" launcher? 
I'm not an advanced user, but let me know what I can do to help.

**Comment by krompus** [comment]
@drcasser Thanks for confirming! Yep, Steam (Runtime). I'm afraid the issue persists after applying those launch options. Glad it's worked for you though!

**Comment by HonkingGoose** [comment]
Hello @doitsujin, thanks for your work on dxvk! :smile: What amount of VRAM would be the recommended amount then? 

I don't see the amount of VRAM mentioned anywhere on these pages:
https://github.com/doitsujin/dxvk/wiki
https://github.com/doitsujin/dxvk
It doesn't seem to be mentioned on this Dark Souls III issue either. 

Maybe a recommended VRAM amount should be mentioned on the dxvk github pages?

I currently have a Nvidia GTX780Ti with 3 GB of VRAM, but I don't know whether this is considered a low VRAM card for the purpose of using DXVK. If I need to buy a newer GPU, it would be very handy to know what the VRAM on that newer card should be. :+1:

**Comment by kisak-valve** [comment]
Hello @ArsenArsen, because this game got whitelisted, please open a new issue for your UI scaling issue so that it can be tracked properly.

**Comment by Brisse89** [comment]
` dxvk.allowMemoryOvercommit = True`

This solved the performance issue when warping between bonfires for me.

**Comment by megatog615** [comment]
> ` dxvk.allowMemoryOvercommit = True`
> 
> This solved the performance issue when warping between bonfires for me.

How exactly are you applying your dxvk config values? Setting the DXVK_CONFIG_FILE in the launch options with a config file that has this didn't solve the problem for me.

**Comment by Brisse89** [comment]
@megatog615 
In a terminal:
`echo "dxvk.allowMemoryOvercommit = True" > ~/dxvk.conf`

Then in the DSIII launch options in Steam:
`DXVK_CONFIG_FILE=~/dxvk.conf %command%`

This seems to have fixed the problem in my case using a 4GiB R9 Fury graphics card.

**Comment by MatejSpindler** [comment]
> ` dxvk.allowMemoryOvercommit = True`
> 
> This solved the performance issue when warping between bonfires for me.

This also solves performance issue for me on GTX 960 2G

**Comment by Jigoku** [comment]
> > ` dxvk.allowMemoryOvercommit = True`
> > This solved the performance issue when warping between bonfires for me.
> 
> How exactly are you applying your dxvk config values? Setting the DXVK_CONFIG_FILE in the launch options with a config file that has this didn't solve the problem for me.

Same here, this has not affected anything for me. I'm using a gtx 1060 (3gb).

**Comment by doitsujin** [comment]
This workaround may not work on Nvidia, it depends on whether the driver lets you allocate more device-local memory than your card actually has. It's more of a hack anyway.

**Comment by megatog615** [comment]
Yeah that definitely doesn't work for me. GTX 960 here. Your instructions were pretty much how I did it the first time.

> 
> This also solves performance issue for me on GTX 960 2G

How many times did you try warping? It seemed to work fine the first warp, but the second dropped my fps to 8.

**Comment by doitsujin** [comment]
This should be fixed in DXVK as of doitsujin/dxvk@ad5688764c3cdb4649fcc59804c2367a38051f44. 
Here's a build: [dxvk-master.tar.gz](https://github.com/ValveSoftware/Proton/files/2600126/dxvk-master.tar.gz)

**Comment by Brisse89** [comment]
> This should be fixed in DXVK as of [doitsujin/dxvk@ad56887](https://github.com/doitsujin/dxvk/commit/ad5688764c3cdb4649fcc59804c2367a38051f44).

Sorry if this is off topic, but I would like to test and see if it works but I am encountering errors while trying to build DXVK. I've never done it before so I'm probably just doing some silly mistake.

[Build log](https://gist.github.com/Brisse89/dbb1804b0f42bf49d7a399e67f9e0548)

From the log I would guess that I am missing a dependency, which I guess would be [libwine-development-dev](https://packages.debian.org/sid/libwine-development-dev) but the thing is that It's already installed so I must be doing something wrong.

**Comment by doitsujin** [comment]
Build requirements and the error you are experiencing are documented [here](https://github.com/doitsujin/dxvk#troubleshooting), but that's off-topic to this issue.

I updated my prebuilt comment with pre-built binaries.

**Comment by Brisse89** [comment]
Applied it to Proton 3.16-4 beta and it seems to have fixed the bonfire warping issue.

**Comment by megatog615** [comment]
When will the patch make it into official Proton builds for Steam?

**Comment by indeedwatson** [comment]
I'm not sure if I'm doing something wrong, because it seems to work for everyone else, but I tried everything from here and nothing is working.

I copied and overwrote the files in `~/.steam/steam/steamapps/common/Proton 3.16/dist/lib/wine/dxvk/` with the ones that @doitsujin just posted. I'm running arch, i5 4590 cpu and AMD R9 270x GPU.

In game normally I get around 30-40fps with min graphics at 1080p (in windows I got 50-60fps), but when I die it drops, sometimes to ~20fps (and then if I die again, down to <10fps) or directly to <10fps.

**Comment by megatog615** [comment]
You overwrote the 32-bit libs and forgot the 64-bit ones. Dark Souls 3 is 64-bit.

**Comment by doitsujin** [comment]
@indeedwatson you need to replace the files in `dist/lib64/wine/dxvk` as well.

**Comment by indeedwatson** [comment]
I knew it was an error on my part, thank you for spotting that, it seems to work now :)

**Comment by arendjr** [comment]
I'm using SteamOS and when I try to launch the game I see a white screen for about 20 seconds and then I suddenly get thrown back to Steam, so presumably the game crashed. The machine has an Intel i5 processor and an NVidia 1060 GPU and is hooked to the TV with HDMI. I'm using a Steam Controller with it.

I can log into it with ssh, so if I can gather some additional info that would be valuable I'd be happy to help.

**Comment by michaeltintiuc** [comment]
@Emanem since you're the only one that posted his resolution and it's ultra-wide, are you getting side borders or is it truly fullscreen? none of the ultra-wide resolutions really work for me

**Comment by Emanem** [comment]
@michaeltintiuc I think I get the black borders, but since a recent release and recent drivers, I get 60 FPS stable all the way.

**Comment by robbert-vdh** [comment]
It looks like this has something to do VRAM not being freed properly. I've been playing the game on a GTX 970 at 1440p and while the game initially runs at a smooth 60 FPS, the frame rate will drop to either 32 or 12 FPS after dying or warping to another bonfire. This gets paired with an increase in VRAM usage from 67% to 97% or higher. That leads me to believe that either the game or (hopefully) DXVK is leaking memory by not freeing resources properly. This combined with the fact that the GTX 970 had that weird 3.5 + 0.5 GB VRAM setup with the last 0.5 GB being slower memory would explain the slowdowns at higher resolutions. For some reason though this behaviour even occurs when the game is loading the same area (i.e. when dying right next to a bonfire).

**Comment by Brisse89** [comment]
@robbert-vdh It should be fixed by now, at least on the 3.16-branch. If you're still at 3.7 then try switching to 3.16.

**Comment by robbert-vdh** [comment]
~~@Brisse89
It's definitely still happening for me on the 3.16-6 beta! Though I did notice just now that DXVK_HUD somehow reports that its running version 0.90 and not 0.94 like it's supposed to. I'll switch to the stable branch to see if the problem persists there.~~

I've wiped `<steamapps>/common/Proton 3.16 Beta` and somehow everything's working again like it should! I ran a quick `fd d3d11.dll --type f --print0 | xargs -0 md5sum` just before that and it looked like the warframe-linux script had somehow managed to overwrite Proton's own DXVK library with an older version. Thanks!

**Comment by tonijarjour** [comment]
Game does not start at all. Have tried with both Nvidia and Nouveau drivers.

**Comment by Emanem** [comment]
> Game does not start at all. Have tried with both Nvidia and Nouveau drivers.

@tonijarjour please print the logs, mate - the game runs perfectly for all of us...

**Comment by kisak-valve** [comment]
Hello @tonijarjour, please open a separate issue report.

**Comment by xexpanderx** [comment]
> Gets to a point where I get a mouse cursor then exits without an error

Exactly the same problem!

**Comment by KorinFlakes** [comment]
After updating to Ubuntu 18.10 Cosmic Cuttlefish I've been getting really bad stuttering on any version of Proton (even though the FPS is 60). When I restart my PC after playing the game I get a warning saying Buffer Overflow detected and it points to the Dark Souls III exe file.

Not sure why this started after upgrading my Ubuntu but would love a fix for it. Sick of restarting my PC just to stop it from lagging.

**Comment by HonkingGoose** [comment]
>When I restart my PC after playing the game I get a warning saying Buffer Overflow detected and it points to the Dark Souls III exe file.

@KorinFlakes I've been having buffer overflow messages on exit for all Dark Souls games (except maybe Dark Souls Remastered, I don't have that game, so I can't test that game.).

My buffer overflows just happen on exit, they don't seem to hamper anything, as far as I can tell. But you can look at my report, and maybe verify my findings? That way at least we know the buffer overflow is not caused by my specific configuration.

My report can be found here: #1962

**Comment by hhyyrylainen** [comment]
Crashes on new game for me as well. Renaming `fdp_opening.bik` works. After getting past that I'm getting quite a bit of lag.

- GPU: GTX 1080 Ti
- Driver/LLVM version: nvidia 418.56
- Kernel version: 5.0.6-200
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/hhyyrylainen/b37480d424240a4c72a0d1a2a3fbd1a2
- Proton version: 3.16-4 (recommended)

Proton log file: https://gist.github.com/hhyyrylainen/8801236fe770a3ea38f6da72959e5a01.

**Comment by kisak-valve** [comment]
Hello @hhyyrylainen, please open a separate issue report.

**Comment by vegerot** [comment]
> @michaeltintiuc do you have nvidia-driver-396, i had the same issue but updating the nvidia driver fixed it for me. (with nvidia-390 i got the same crash)

How did you fix this?

**Comment by deathxxx123** [comment]
Game will run with
WINEDLLOVERRIDES="xaudio2_7=n,b" %command%
Or without this command, you need delete or rename more videos than only opening one. Because every video in gameplay will soft-lock you'r progress. Just like after battle with Yorm The Giant, you cannot proceed to Grand Archives, because the video(https://youtu.be/pOr0ab_EDnw?t=611) after the fight cannot be played, and you will be not teleported after end, because you'r hero is a part from that video. It seems that, videos are boud to game progress, and cannot continue without game think that video are played. You can play, but cannot proceed to Grand Archives. Also videos from Dark Sous III, are different from other games. In videos you'r real player are in there.

**Comment by kisak-valve** [comment]
Hello, please retest this game with Proton 6.3-6 and see if any regressions have crept in since this issue was last evaluated.

**Comment by Brisse89** [comment]
Played almost four hours this evening, from the start of the game to "Road of Sacrifices". I don't see any new performance issues. Performance isn't perfect however. I need to keep "Shadow quality" set to high or lower (max causes severe performance drops in some areas), and "Shader quality" to medium or lower, otherwise there will be some areas that drop under 30fps. All other settings I can max out no problem and still be able to keep a steady 60fps. 

Strangely however, performance seems to drop after a while, and after playing for an hour or so the frame-rate will hover around 40-50fps instead of 60fps like it does just after starting the game. Quitting and restarting brings it up to 60 again. This is not unique for Proton 6.3-6 however, it's just the same when using 3.16. If anything I'd say 6.3-6 might be slightly better optimized. Overall the game is working really well with either of these Proton versions.

Tested on Debian testing, using GNOME 40.4 (Wayland), Mesa 21.2.0 and Linux 5.10, using a Ryzen 2700X, R9 Fury and 32GiB of RAM.

**Comment by Brisse89** [comment]
Just played another hour or so and now I ran into the bonfire warping issue where I was getting a smooth 60fps, then warped and performance suddenly fell to 2.3fps and I got the "Frame rate unsuitable for online play" message before being kicked back to the main menu. This was supposedly fixed by https://github.com/doitsujin/dxvk/commit/ad5688764c3cdb4649fcc59804c2367a38051f44 but the issue might have returned. It's only happened to me once on 6.3-6 so far. I'll keep testing and let you know if it happens again.

**Comment by Brisse89** [comment]
After 15h 20m playtime, it happened again, now on Proton 3.16-9. 

![Skärmbild från 2021-09-04 18-56-25](https://user-images.githubusercontent.com/10871070/132102534-e10a2b28-5e91-4fdd-8ae1-fbb2bff50453.png)

It is much less frequent than it used to be, but clearly it isn't fully fixed in either of these Proton releases.

As far as differences between 3.16 and 6.3, I noticed the latter performs poorly in the area called "Archdragon Peak". This is why I happened to be running 3.16 at the time I ran into the issue above. It performed noticeably better in this area of the game. 6.3 would run around ~30fps, and lowering the graphics settings didn't do anything to change that.

**Comment by hhyyrylainen** [comment]
I didn't have time to play for long, but I think I can at least say that the low performance problems are mostly gone (and I didn't have to do the video file renaming to be able to start a new game). I played for almost half an hour and only noticed about 5 or so slight lag spikes, other than that I got a steady 60 FPS. I used Proton 6.3-6. Or it could be that I now have different GPU (switched from nvidia to AMD) and CPU which fixed the performance. 
One serious problem I had was that the game wouldn't let me select 4K resolution (I have 2 4K and 2 1440p screens). it only allowed me to select up to 2560x1440 resolution, even though I was able to move the game away from my primary monitor to one of the 4K ones by going to windowed mode and going back to fullscreen after moving the window.

**Comment by Brisse89** [comment]
It happened again so that's three times during this play-through. I've logged 23h34m and almost completed the entire game completionist style, including DLC's, sans three of the last bosses. I've been to every area of the game and I can't say that I can see a performance difference in any other area than already mentioned "Archdragon Peak".
